### PR TITLE
[DEV] use latest jupyterlab APIs

### PR DESCRIPTION
**Pull request by minrk**
updates everything except for webpack 5, which doesn't seem to work yet. This uses the packages from the latest jupyterlab 3.0 release candidate.

The changes are relatively small:

- use `KernelManager().startNew` instead of `Session.startNew`
- `appendToken` behavior is now opt-in instead of the default. This is required for cross-origin use cases like ours
- no longer accepts partial serverSettings, only filled-out by makeSettings

This seems to work just fine, but I don't want to introduce any conflicts with what other sprinters are working on

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -32,21 +32,23 @@ module.exports = {
...
+    // shim(/@lumino\/coreutils\/lib\/random/),
+    // shim out some unused lumino
     shim(
-      /@phosphor\/widgets\/lib\/(box|commandpalette|contextmenu|dock|grid|menu|scroll|split|stacked|tab).*/
+      /@lumino\/widgets\/lib\/(box|commandpalette|contextmenu|dock|grid|menu|scroll|split|stacked|tab).*/
     ),
-    shim(/@phosphor\/collections\/lib\/(bplustree).*/),
-    shim(/@phosphor\/(dragdrop|commands).*/),
+    shim(/@lumino\/collections\/lib\/(bplustree).*/),
+    shim(/@lumino\/(dragdrop|commands).*/),
```
Comment (minrk) [inline webpack.config.js:44]: The shims will need revisiting, as the bundle got even bigger again!

**Comment by minrk** [comment]
This is working, and should no longer conflict with sprint contributions

**Comment by choldgraf** [comment]
@minrk will this be an issue if JupyterLab isn't installed on the particular binder instance that a person requests? E.g. I think that Binder ships with JupyterLab 2.X right now, and I dunno when it'll switch to 3.x

**Comment by minrk** [comment]
It shouldn't. This only means the jupyterlab *javascript* APIs that are pulled in as part of thebe itself. It should not be sensitive to the server-side version.

**Pull request merged**